### ISSUE TITLE
isLoaded not firing as expected

### ISSUE THREAD

**Issue by ecramer** [realm/realm-java#6556]
#### Issue
`isLoaded()` is no longer being fired as expected. 

#### Actual Results
isLoaded never returns true in some cases. 

#### Steps & Code to Reproduce

It doesn't happen all the time but we can definitely get into a state where isLoaded never returns true. In the below, `realmObject.isLoaded()` returns false and then never returns true again. We have verified that the Single isn't being disposed of before it has a chance to return true. There is no error being thrown.

```
return realmObjectExample
				.<T>asFlowable()
				.filter(realmObject -> realmObject.isLoaded())
				.firstOrError()
```

The object definitely exists in the database. 
It happens on individual realm objects and RealmResults.
This works in version 5.8.0 and 5.9.0 which we are currently stuck on because of this. 

#### Version of Realm and tooling

Realm version(s): 5.10.0 and above

Realm Sync feature enabled: No

Android Studio version: 3.4.1

Android Build Tools version: 3.4.1

Gradle version: gradle-5.1.1

Which Android version and device(s): Pixel 2, Android 9

**Comment by cmelchior** [comment]
Hmm, weird. It could indicate that either the Realm was closed or the Realm Object was GC'ed, but the Flowable should keep a strong reference to it, so as long as you have a reference to the Flowable it should work.

Are you able to create a minimal working project that demonstrates the behavior?

**Comment by ecramer** [comment]
I haven't been able to create a minimum working project for this issue but I have done more debugging in our application. 

OsResults -> `notifyChangeListeners` is not getting called at all in this circumstance.

The listener on the Single is definitely still active and open. If a network request happens in the background while this screen is open and waiting, isLoaded will fire and come back as true when that background transaction is complete.

**Comment by ecramer** [comment]
I haven't been able to create a minimum working project to recreate the behaviour but it is still 100% reproducible in our project.

I did some further debugging and have found that `OsResults.notifyChangeListeners` is not being called in this situation at all. 
I have confirmed that Realm has not been closed or GC'd as well. If a network request fires resulting in some writes on a background thread, the Single triggers once that write transaction has completed. 

Further I have found that if I break point in findFirstAsync and step through the execution of the PendingRow constructor, it will work. This is leading me to believe that we are dealing with a race condition (which is also probably why I can't reproduce it in a sample project). Is any of this helpful?
